app/services/register_user_account.rb

```
@@ -20,15 +20,17 @@ def run
     Honeycomb.start_span(name: 'register_user_account.run') do |span|
       span.add_field('app.salesforce.contact.id', @salesforce_participant.contact_id)
 
-      # Need to have the User record saved before the Canvas sync runs since it relies on it.
-      # Essentially do an upsert
-      @new_user = User.find_or_create_by(
+      # Local user record was already created during Sync from Salesforce.
+      # If we can't find it, something's wrong; raise immediately.
+      @new_user = User.find_by!(
         salesforce_id: @create_user_params[:salesforce_id]
       )
```

Sync to LMS runs nightly. A lot of times in the mad-dash during launch, folks are being recruited last minute (like they show up at the first event b/c they heard about Braven). I think we need to still create their local user record if it doesn't exist. Is there any downside to that with this new flow (which I'm still wrapping my head around)?

---

We used to rely on Sync To LMS and staff would run it over and over again manually during this mad dash to get everyone access to the Portal

---

L48 in the registrations controller would have to change. I think it's fine to allow enumeration of the sign-up ID if we use the more secure (longer, random) token we were talking about instead of the SF ID.

---

Hmm, we'd have to think it through and make sure that wouldn't hide any errors. I think it'd be ok?

---

this is now addressed.